File: Spring/Task 2 (Hierarchy)/sources/ice_cream/milky_ice_cream/shaved_ice/shaved_ice.cpp
//
// Created by rinsler on 30.03.18.
//

#include "shaved_ice.h"

ShavedIce::ShavedIce(
        std::string name,
        std::string manufacturer,
        Ingredient flavor,
        Ingredient fruitIce,
        Ingredient glaze,
        Ingredient syrup
) :
        MilkyIceCream(name, manufacturer, flavor, glaze, syrup),
        fruitIce(fruitIce)
{

}

std::string ShavedIce::getRecipe()
{
    return MilkyIceCream::getRecipe() +
           " Add " + std::to_string(fruitIce.weight) + " grams of frozen " + fruitIce.name + " juice.";
}

File: Spring/Task 2 (Hierarchy)/src/icecream/ShavedIce.java
package icecream;

import com.sun.istack.internal.Nullable;

public class ShavedIce extends MilkyIceCream {
    protected Ingredient fruitIce;

    public ShavedIce(String name, String manufacturer, Ingredient flavor, Ingredient fruitIce, @Nullable Ingredient glaze, @Nullable Ingredient syrup) {
        super(name, manufacturer, flavor, glaze, syrup);
        this.fruitIce = fruitIce;
    }

    @Override
    public String getRecipe() {
        return super.getRecipe() +
                " Add " + fruitIce.getWeight() + " grams of shaved frozen " + fruitIce.getName() + " juice.";
    }
}
